pkgs/tools/networking/linux-router/default.nix

```
@@ -0,0 +1,102 @@
+# SOURCE: https://github.com/NixOS/nixpkgs/pull/92787/files
+
+{ stdenv, lib, pkgs, fetchFromGitHub
+
+# Dependencies
+, bash , procps , iproute2 , dnsmasq , iptables
+
+# WiFi hotspot dependencies
+, hostapd , iw
+, wirelesstools ? null # for iwconfig
+, haveged ? null
+, qrencode ? null
+
+# General dependencies
+, coreutils , flock , gawk , getopt , gnugrep , gnused , which
+, networkmanager ? null
+
+# To fall back to haveged if entropy is low.
+#
+# Defaulting to false because not having it does not break things.
+# If it is really needed, warnings will be logged to journal.
+, useHaveged ? false
+
+# nmcli is not required for create_ap.
+# (defaulting to true because it is very likely already present)
+, useNetworkManager ? true
+
+# You only need this if 'iw' can not recognize your adapter.
+, useWirelessTools ? true
+
+# You only need this if you wish to show WiFi QR codes in terminal
+, useQrencode ? true
+}:
+
+assert useHaveged        -> !isNull haveged;
+assert useNetworkManager -> !isNull networkmanager;
+assert useWirelessTools  -> !isNull wirelesstools;
+assert useQrencode       -> !isNull qrencode;
+
+with lib;
+
+let
+  binPath = makeBinPath ([ iw procps hostapd iproute2 getopt bash dnsmasq
+    iptables coreutils which flock gnugrep gnused gawk ]
+    ++ optional useHaveged        haveged
+    ++ optional useNetworkManager networkmanager
+    ++ optional useWirelessTools  wirelesstools
+    ++ optional useQrencode       qrencode);
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "linux-router";
+  version = "0.6.2";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "garywill";
+    repo = "linux-router";
+    rev = "${version}";
+    sha256 = "193bnlwmjxsk0cri6xdylf218qayldn02pdnppvbd39ls361776z";
+  };
+
+  nativeBuildInputs = [ pkgs.makeWrapper ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin/ $out/.bin-wrapped
+    mv lnxrouter $out/.bin-wrapped/lnxrouter
+    makeWrapper $out/.bin-wrapped/lnxrouter $out/bin/lnxrouter --prefix PATH : ${binPath}
```

```suggestion
    wrapProgram $out/bin/lnxrouter --prefix PATH : ${binPath}
```
this takes care of moving the original binary to a different name.

---

Thanks, I had that before but the `--help` page uses the executable name to infer the name of the tool and when using `wrapProgram` it displays `.linux-router-wrapped` in the usage examples (see https://github.com/NixOS/nixpkgs/pull/137133/commits/5410b55ae7b808a83fd5196bd668e77ea3bfb095)

---

OK, I see. However, I do not understand why it happens as `makeWrapper/wrapProgram` tries to avoid that setting the name.

---

`wrapProgram` seems to rename the original executable to `.<exe-name>-wrapped` and then create a script with the original `<exe-name>` that sets up the environment and then `exec`s the `.<exe-name>-wrapped`.

The `linux-router` executable (a bash script) uses it's exe name for the `--help` text. See the first and last line here:

https://github.com/garywill/linux-router/blob/c2b21bb3913c6f775893fe1bfedffaa8dcd39b03/lnxrouter#L5-L17

```bash
PROGNAME="$(basename $0)"

export LC_ALL=C

SCRIPT_UMASK=0122
umask $SCRIPT_UMASK

usage() {
    cat << EOF
linux-router $VERSION (https://github.com/garywill/linux-router)
Released under LGPL, with no warranty. Use on your own risk.
Usage: $PROGNAME <options>
```

The `$0` contains the path to the script file which is renamed by `wrapProgram` so that it is now `/nix/store/<hash>-linux-router-<version>/bin/.linux-router-wrapped`. Then `basename` then gets the `.linux-router-wrapped` name which is then used in the last line for the usage example.

---

The wrapper calls the orginal with `exec -a <originame>`, which should take of it by effectively changing `$0` to the orginal name (not sure why it does not in that setting).